cmd/dagger/project.go

```diff
@@ -84,35 +83,39 @@ var addLocalCmd = &cobra.Command{
 var addGitCmd = &cobra.Command{
 	Use: "git",
 	RunE: func(cmd *cobra.Command, args []string) error {
+		ctx := context.Background()
 		cfg, err := getThisProjectConfig()
 		if err != nil {
 			return err
 		}
 
-		startOpts := &engine.Config{
-			Workdir:     workdir,
-			ConfigPath:  configPath,
-			SkipInstall: true,
```

Comment:
I think we dropped the `SkipInstall` here unless I missed something. If we can keep it that would be preferable because the only thing that's needed here is to load project metadata; there's no need to actually compile the extension (and fail if it has an error).

The only other possibility would be to move loading of project out from engine, but that's the whole other can of worms we talked about before, not worth it yet probably.

This isn't blocking if it's time consuming, but worth an issue if not fixed here.

Reply:
Yep, I was actually puzzled by this.

If I understand correctly, setting SkipInstall would only load the project under ctx.Project, which is not exposed anymore anyway, correct?

So a similar behavior could be achieved with a WithConfigPath(""), or to make it more explicit a better name would be WithNoExtensions()?

I see this code is querying for its own loadExtension so it doesn't care that the engine is actually loading the extensions (and skipping install)

Reply:
Ah yeah, I forgot the code that I wrote several days ago already 😵‍💫😵‍💫😵‍💫 This file actually already is doing the project load itself (line 192), so the only thing needed is for the engine to not load anything, which will be the implicit behavior if you don't set ConfigPath, so actually the only change here should be to just not call `WithConfigPath` at all.

Reply:
I pushed a small update in a separate commit `engine: remove redundant functionality` -- helps clarify the load project stuff in engine (much smaller surface area).

How does that look? Can easily revert

Reply:
(also can add an explicit `dagger.WithNoExtensions()`)

Reply:
That commit looks good overall, but I don't think you need `NoExtensions` even. AFAICT if you don't provide a ConfigPath, that should just be the same thing. I think that makes sense: no extension config, no extensions loaded.

Then special cases (like this project command) that need different extension loading logic (like skipping install) can just do the call themselves.

Reply:
Ah, actually we call `NormalizePaths` in `engine.Start`, so this doesn't quite work as is (it will see empty ConfigPath and then default it to `$(pwd)/dagger.json`). I actually think that in and of itself should be refactored, but that's getting way out of scope here.

How about we just leave `NoExtensions` for now (it's an obscure thing, only needed by this project command) and then we can refactor this a bit more in the future. I think I see a path for how to do it cleanly, I'll make an issue.

Reply:
Made an issue here: https://github.com/dagger/dagger/issues/3375

Reply:
Done in last commit